lib/vector-core/src/event/value.rs

```diff
@@ -160,15 +162,13 @@ impl From<serde_json::Value> for Value {
     fn from(json_value: serde_json::Value) -> Self {
         match json_value {
             serde_json::Value::Bool(b) => Value::Boolean(b),
-            serde_json::Value::Number(n) => {
-                if let Some(i) = n.as_i64() {
-                    Value::Integer(i)
-                } else if let Some(f) = n.as_f64() {
-                    Value::Float(f)
-                } else {
-                    Value::Bytes(n.to_string().into())
-                }
-            }
+            serde_json::Value::Number(n) => n.as_i64().map_or_else(
+                || {
+                    n.as_f64()
+                        .map_or_else(|| Value::Bytes(n.to_string().into()), Value::Float)
+                },
+                Value::Integer,
```

Comment:
Nit: I'd find this a bit easier to read keeping the `or_else`s at the top level.

I think something like:

```
n.as_i64().map(Value::Integer)
  .or_else(|| n.as_f64().map(Value::Float))
  .or_else(|| Value::Bytes(n.to_string().into())
```

Reply:
Yeah, I like that form. Clippy's mechanical translations can be... weird.

Reply:
The types on your suggested form don't quite work and the more I look at clippy's suggested version the more I find it difficult to read.

Reply:
I adjusted this in cbefc3d. It reads a little more clear to me but I'm tempted to just go back to the way it was. Clippy might not like the original but of the options that have worked out so far it's the one I like the most.